### ISSUE TITLE
Enable screen recording for all tests by default

### ISSUE THREAD

**Issue by ubuntudroid** [mobile-dev-inc/Maestro#2053]
### Use case

Screen recording can be a super helpful thing to debug why a test failed. However, right now one needs to manually activate them for every flow by using the `startRecording` and `stopRecording` commands which leads to boilerplate.

### Proposal

Provide a way to toggle screen recordings for all flows, e.g. via a command line parameter `--enableScreenRecording`. The names of the mp4 files could be automatically deduced from the test file, e.g. `test1.yaml` -> `test1.mp4`.

As an extension of this, and potentially even more useful, there could also be another flag which would enable storing of recordings just for failed tests, such as `--enableScreenRecordingOfFailedFlows`, to help preserve system storage.

**Comment by siddhantsoni** [comment]
Any update on supporting this functionality? This seems super useful and necessary for test debugging and test report generation and can be supported out of the box by maestro

**Comment by edwinvrgs** [comment]
Any updates on this?

**Comment by Fishbowler** [comment]
Doesn't seem like anyone has picked it up, but now that you've commented, you'll get updates if someone fancies it.

**Comment by Fishbowler** [comment]
I wonder what the gap is between this and the existing `maestro record` functionality?

**Comment by ubuntudroid** [comment]
> I wonder what the gap is between this and the existing `maestro record` functionality?

`maestro record` is limited to 2 minutes. And I'm not sure you can pass multiple yamls. Also, I don't know whether it supports all flags supported by plain `maestro test`.

**Comment by Fishbowler** [comment]
`maestro record --local` isn't limited to 2 minutes.

If we're missing some flags, we should fix that. I'd expect `record` to be a synonym of `test`, but with recording.

**Comment by ubuntudroid** [comment]
> maestro record --local isn't limited to 2 minutes.

The [documentation](https://docs.maestro.dev/maestro-flows/workspace-management/record-your-flow#how-to-record) seems to suggest otherwise:

> It's important to notice that recordings are limited to a maximum of two minutes. If your Flow runs longer, the recording will stop automatically.

---

> I'd expect record to be a synonym of test, but with recording.

Yes, exactly! In the meantime, we have workarounds in place - so no pressure from my end and thanks for all the efforts you're putting into Maestro!

**Comment by Fishbowler** [comment]
Indeed, see https://github.com/mobile-dev-inc/maestro-docs/issues/194

**Comment by Fishbowler** [comment]
Agree.

`maestro record` records a Flow. `maestro test` does loads more - tag selection, sharding, CI reports

This should be a flag on `test`, so that recording picks up the rest of what `test` already does rather than being a separate way in.

Suggest `--record`, writing into the run's existing artifact directory alongside screenshots and logs. Those have improved considerably since this was logged, so the naming scheme proposed above (`test1.yaml` → `test1.mp4`) probably isn't needed. The artifact directory already gives recordings a per-run home.

With that in place, `maestro record` can be deprecated entirely rather than maintained as a second entry point.
